cantools/database/utils.py

```diff
@@ -86,6 +87,15 @@ def _encode_fields(fields: Sequence[Union["Signal", "Data"]],
 
             unpacked[field.name] = _transform(value)  # type: ignore[operator]
             continue
+        elif isinstance(field, decimal.Decimal) and isinstance(field, Signal):
```

Comment:
Should probably be `isinstance(value, decimal.Decimal)` instead. This condition will never trigger otherwise